lib/xcodeproj/project/object/group.rb

```diff
@@ -437,10 +438,9 @@ def sort(options = nil)
                 next groups_position == :above ? 1 : -1
               end
             end
-
-            result = File.basename(x.display_name.downcase, '.*') <=> File.basename(y.display_name.downcase, '.*')
+            result = XcodeSortString.new(File.basename(x.display_name, '.*')) <=> XcodeSortString.new(File.basename(y.display_name, '.*'))
```

Comment:
should `XcodeSortString` internally handle sorting lexicographically by basename and then extname ?

Reply:
No, I think this situation is just to handle identical filenames in different folders: let's keep the fallback independent from the sort helper..

Reply:
I've tested this situation with Xcode, and Xcode does not sort at all when filenames are identical: it will simply keep the existing order. That supports that we shouldn't include this behavior in `XcodeSortString` itself, and we should simply keep it externally like that.

Reply:
@segiddins actually, I misunderstood `extname` for "extended name" (aka fullpath) when actually it's a function to get the "file extension". So yes, you're right, there is something to fix here. I'm looking at it.